### PR TITLE
refactor(frontend): ensure proper validation in organization add credits modal (org support)

### PR DESCRIPTION
**Pull request by hieptl**
## Summary of PR

**Success criteria:**
- Placeholder.
- Minimum values.
- If the input is invalid, render an error message.

We can refer to the video below for the output of the pull request.

https://github.com/user-attachments/assets/f17c1bb5-9b78-4f3f-b070-97bc623b6892

## Change Type

- [ ] Bug fix
- [ ] New feature
- [ ] Breaking change
- [x] Refactor
- [ ] Other (dependency update, docs, typo fixes, etc.)

## Checklist

- [x] I have read and reviewed the code and I understand what the code is doing.
- [x] I have tested the code to the best of my ability and ensured it works as expected.

## Fixes

Resolves #(issue)

## Release Notes

- [ ] Include this change in the Release Notes.

---

To run this PR locally, use the following command:

GUI with Docker:
```
docker run -it --rm   -p 3000:3000   -v /var/run/docker.sock:/var/run/docker.sock   --add-host host.docker.internal:host-gateway   -e SANDBOX_RUNTIME_CONTAINER_IMAGE=docker.openhands.dev/openhands/runtime:0d807dc-nikolaik   --name openhands-app-0d807dc   docker.openhands.dev/openhands/openhands:0d807dc
```

### CODE REVIEW

File: frontend/__tests__/routes/manage-org.test.tsx [hunk 1]
```diff
@@ -162,6 +162,511 @@ describe("Manage Org Route", () => {
...
+
+        await waitFor(() => {
+          const errorMessage = screen.getByTestId("amount-error");
+          expect(errorMessage).toHaveTextContent(
+            "PAYMENT$ERROR_MAXIMUM_AMOUNT",
+          );
+        });
+      });
+
+      it("should display error message for decimal values", async () => {
```
Comment (amanape) [inline frontend/__tests__/routes/manage-org.test.tsx]: Hmm, I haven’t thought of this before, but I think we do want to support decimal values. Unless we don’t?

Comment (hieptl) [reply frontend/__tests__/routes/manage-org.test.tsx]: Hello @amanape,

At the moment, decimal values are not supported.

We currently use a function named "amountIsValid", which is invoked in payment-form.tsx on the current Billing page to validate the input amount. Based on the logic implemented in this function, decimal values are not permitted.

For additional details, please refer to the following link:

https://github.com/OpenHands/OpenHands/blob/d7b36c9579661069e685fd56334bcb58f5cb98af/frontend/src/utils/amount-is-valid.ts#L10

Thank you! 🙏

**Comment by hieptl** [comment]
> Let’s enable the button ALWAYS and validate / render the error message AFTER the user attempts to submit the form. For example:

> It also looks like the describe blocks "Real-time Validation” and "Edge Cases” are duplicate tests already covered in previous tests. We should avoid duplicating tests or test logic.

Hello @amanape,

I have updated the code based on your feedback.

Thank you for your feedback! 🙏
